backend/models/pin-services.js

```diff
@@ -84,6 +85,43 @@ async function downvotePin(id) {
   }
 }
 
+async function addCommentToPin(pinID, comment) {
+  let pin = await findPinById(pinID);
+  try {
+    const commentToAdd = new commentModel(comment);
+    const savedComment = await commentToAdd.save();
+    pin.comments.push(commentToAdd._id);
+    const savedPin = await pin.save();
+    return savedPin;
+  } catch (error) {
+    console.log(error);
+    return undefined;
+  }
+}
+
+async function removeCommentFromPin(pinID, commentID) {
+  let pin = await findPinById(pinID);
+  try {
+    pin.comments.pull(commentID);
+    const savedPin = await pin.save();
+    return savedPin;
+  } catch (error) {
+    console.log(error);
+    return undefined;
+  }
+}
+
+async function getPinComments(id) {
```

Comment:
Can you change this to return a list of json objects rather than a list of commentIds?

Reply:
You might be able to just change line 117 to be
pin = await pinModel.findById(id).populate('comments');

I recommend using MongoDB's population functionality: [](https://mongoosejs.com/docs/populate.html)